### DISCUSSION TITLE
Multiple tabs introduces bottom padding when `window-step-resize = true`

### DISCUSSION THREAD

**Discussion by JakeWharton** [ghostty-org/ghostty#2945] [Feature Requests, Ideas]
I'm on macOS 14.7.1 (23H222) with Ghostty 9f75d93a.

My full config is
```
initial-window = false
copy-on-select = true
window-save-state = always
background-opacity = 0.9
background-blur-radius = 5
window-step-resize = true
keybind = global:alt+grave_accent=toggle_quick_terminal
mouse-hide-while-typing = true
macos-option-as-alt = true
```

Opening a new window and pressing <kbd>Enter</kbd> a bunch puts my cursor right up against the bottom of the window, as expected:

![Image](https://github.com/user-attachments/assets/3300170e-db6c-412b-b12f-748f004d0bd9)

Starting a new tab and doing the same thing yields some padding at the bottom, presumably because the tab bar height is unlikely to be a multiple of the character height and the window height is not adjusted in any way:

![Image](https://github.com/user-attachments/assets/48f38eaa-49c8-473b-b9f6-0330caeaa4bd)

Closing all tabs but one which causes the tab bar to disappear puts the cursor back at the bottom.

Resizing with multiple tabs does not fix the extra bottom padding:

![Image](https://github.com/user-attachments/assets/336c468b-84a6-4d33-8281-443c26f20e7b)

**Comment by mitchellh** [comment]
I swear another discussion already existed around this but I can’t find it. I think there is a general question and discussion to be had around what the behavior of step resize should be when native UI elements modify the viewport such as a tab bar.

**Reply by JakeWharton** [reply]
I didn't see any others. There's https://github.com/ghostty-org/ghostty/pull/2919 which I've been following, but I don't have that specific problem.

I would think that the two reasons to enable step resize is that you want one or both of:
- Last line is at the bottom of the window
- Last character is at the right of the window

**Comment by mitchellh** [comment]
Related issue created (doesn't resolve this issue): #5299

**Comment by kd3n1z** [comment]
+1 on this! Ghostty 1.1.3

**Comment by AdiCahyaSaputra** [comment]
https://github.com/user-attachments/assets/8b7a3cb3-8332-4978-bada-99d8e4e0c474

It's also happening on linux.

My ghostty version
```bash
Ghostty 1.3.1-arch2

Version
  - version: 1.3.1-arch2
  - channel: tip
Build Config
  - Zig version   : 0.15.2
  - build mode    : .ReleaseFast
  - app runtime   : .gtk
  - font engine   : .fontconfig_freetype
  - renderer      : renderer.generic.Renderer(renderer.OpenGL)
  - libxev        : io_uring
  - kernel version: 7.0.8-arch1-1
  - desktop env   : other
  - GTK version   :
    build         : 4.22.2
    runtime       : 4.22.4
  - libadwaita    : enabled
    build         : 1.9.0
    runtime       : 1.9.0
  - libX11        : enabled
  - libwayland    : enabled
```

My Config
```bash
background = #000000
background-opacity = 0.75

window-decoration = false
# Alternatively, use this for more control:
# gtk-titlebar = false
gtk-tabs-location = bottom

shell-integration-features = no-cursor
cursor-style = block
copy-on-select = true

window-padding-x = 10
window-padding-y = 10,0

keybind = ctrl+shift+o=unbind
keybind = ctrl+t=new_tab
keybind = ctrl+w=close_tab
keybind = ctrl+shift+n=toggle_fullscreen
keybind = ctrl+alt+j=new_split:down
keybind = ctrl+alt+k=new_split:right

keybind = ctrl+shift+h=goto_split:left
keybind = ctrl+shift+j=goto_split:down
keybind = ctrl+shift+k=goto_split:up
keybind = ctrl+shift+l=goto_split:right
```
